lib/src/data/models/movie_model.dart

```diff
@@ -25,20 +26,21 @@ class MovieModel extends Movie {
       score: (json['vote_average'].toDouble() / percent),
       genres: List<int>.from(json['genre_ids']),
       overview: json['overview'].toString(),
-      cast: CastModel.fromListOfJson(
-        json['cast'],
-      ),
     );
   }
 
   static List<MovieModel> listOfMovies(List<dynamic> list) {
     List<MovieModel> movies = List.generate(
       list.length,
-      (index) => MovieModel.fromJson(list[index]),
+      (int index) => MovieModel.fromJson(list[index]),
     );
     return movies;
   }
 
+  void setCast(List<CastModel> cast){
+    this.cast = cast;
+  }
```

Comment:
you need to made this kind of set?

Reply:
Movie, Genres and in your case Cast are individual calls, that you need to join together to show the complete information of the movie, you can do it, without any kind of set of the object

Reply:
The original model of movie doesn't have a cast parameters, so separate the cast from the movie

Reply:
Fixed